src/Umbraco.Web.BackOffice/Controllers/AuthenticationController.cs

```diff
@@ -629,7 +679,7 @@ private string ConstructCallbackUrl(string userId, string code)
                 });
 
             // Construct full URL using configured application URL (which will fall back to request)
-            var applicationUri = _hostingEnvironment.ApplicationMainUrl;
+            Uri applicationUri = _httpContextAccessor.HttpContext?.Request.GetApplicationUri(_webRoutingSettings);
```

Comment:
```suggestion
            Uri applicationUri = _httpContextAccessor.GetRequiredHttpContext().Request.GetApplicationUri(_webRoutingSettings);
```

Avoid potential null reference exception